src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -458,6 +467,23 @@ private static void AddDefaultVNetResourceNode(
         }
     }
 
+    private static String pageBlobFormat(final String sourceString) throws Exception {
+        /*Page blob must align to 512-byte page boundaries*/
+        int formatStringLength = sourceString.length() % 512 == 0
```

Comment:
Here is another risk.  `sourceString.length()` might not be the best choice since it might contain non-English characters such as Chinese. The `string.length()` is not equal to `bytes[].length` always and Azure actually validates the latter. I once met the similar issue that resulted a failure in upload using `C#`. So I suggest you replace the `sourceString.length()` with `sourceString.getBytes().length`. And please also make sure it's `UTF-8` encoding.